pkg/client/cli/cliutil/exec.go

```diff
@@ -0,0 +1,63 @@
+package cliutil
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"strings"
+
+	//nolint:depguard // TODO: switch this stuff over to dexec
+	"os/exec"
+
+	"github.com/telepresenceio/telepresence/v2/pkg/client/logging"
+)
+
+// Run will run the given executable with given args and env, wait for it to terminate, and return
+// the result. The run will dispatch signals as appropriate for the given platform (SIGTERM and SIGINT on Unix platforms
+// and os.Interrupt on Windows).
+func Run(ctx context.Context, sCmd SafeCobraCommand, exe string, args []string, env map[string]string) error {
+	cmd := exec.CommandContext(ctx, exe, args...)
+	cmd.Stdout = sCmd.OutOrStdout()
+	cmd.Stderr = sCmd.ErrOrStderr()
+	cmd.Stdin = sCmd.InOrStdin()
+	for k, v := range env {
+		cmd.Env = append(os.Environ(), k+"="+v)
+	}
+
+	var err error
+	if err = cmd.Start(); err != nil {
+		return fmt.Errorf("%s: %w", logging.ShellString(exe, args), err)
+	}
+
+	// Ensure that interrupt is propagated to the child process
+	sigCh := signalNotifications()
+	defer close(sigCh)
```

Comment:
This is panicy.  The `os/signal.Notify` machinery will panic if it tries to write to this closed signal.  Need to call `signal.Stop(sigCh)`.